### PR TITLE
DSL2 conversion

### PR DESCRIPTION
**Pull request by ggabernet**
- Pipeline DSL2 conversion
- Updated template

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)
  - [x] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/cellranger_count.nf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+process CELLRANGER_COUNT {
+
+    label 'process_high'
+
+    publishDir "${params.outdir}",
+        mode: 'copy',
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:'') }
+
+    container "streitlab/custom-nf-modules-cellranger:latest"
```
Comment (grst) [inline modules/local/cellranger_count.nf:14]: I thought we had that in the nf core namespace now?

File: modules/local/gene_map.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        name = "${gtf.baseName}"
+        unzip = "gunzip -f ${gtf}"
+    } else {
+        unzip = ""
+        name = "${gtf}"
+    }
+    """
+    $unzip
+    cat $name | t2g.py --use_version > transcripts_to_genes.txt
+    """
```
Comment (grst) [inline modules/local/gene_map.nf:39]:
```suggestion
    """
    zcat -f $gtf | t2g.py --use_version > transcripts_to_genes.txt
    """
```

File: modules/local/samplesheet_check.nf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def get_samplesheet_paths(LinkedHashMap row) {
+    def meta = [:]
+    meta.id           = row.sample
+    meta.single_end   = row.single_end.toBoolean()
+
+    def array = []
+    if (meta.single_end) {
+        array = [ meta, [ file(row.fastq_1, checkIfExists: true) ] ]
+    } else {
+        array = [ meta, [ file(row.fastq_1, checkIfExists: true), file(row.fastq_2, checkIfExists: true) ] ]
```
Comment (grst) [inline modules/local/samplesheet_check.nf:43]: We don't have single end in this pipeline, do we?

**Review by grst — APPROVED** [review]
I added a few comments, but I don't see any major issues!

**Review by apeltzer — APPROVED** [review]
Nothing really found on my side - looks good.

**Comment by grst** [comment]
Seems some tests have been renamed - Can someone with write access please update the required tests for this repo?

**Comment by ggabernet** [comment]
I'll be merging this and work on the next template update in a separate PR

**Pull request merged**